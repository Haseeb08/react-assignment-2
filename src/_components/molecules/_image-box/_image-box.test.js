import React from "react";
import { create } from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ImageBox from "./_image-box";

const image = {
  id: "1",
  name: "React",
  url:
    "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRWXoLhb1QGq1B52D9rNOdWzmIdy0IVkmKk9Q&usqp=CAU",
};
describe("renders gallery image", () => {
  test("Matches the snapshot", () => {
    const input = create(<ImageBox img={image} />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("Matches the snapshot admin", () => {
    const input = create(<ImageBox img={image} isAdmin={true} />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("Edit image", () => {
    const mockFn = jest.fn();

    const { getByTestId, getByRole, getByText } = render(
      <ImageBox
        img={image}
        isAdmin={true}
        handleImageEdit={mockFn}
        handleImageDelete={mockFn}
      />
    );
    const edit = getByTestId("edit-button");
    fireEvent.click(edit);
    fireEvent.change(getByRole("textbox"), { target: { value: "abc" } });
    expect(getByText("Save")).toBeInTheDocument();
    fireEvent.click(getByText("Save"));
    expect(mockFn).toHaveBeenCalledTimes(1);
    fireEvent.click(getByText("Delete"));
    expect(mockFn).toHaveBeenCalledTimes(2);
  });
});
