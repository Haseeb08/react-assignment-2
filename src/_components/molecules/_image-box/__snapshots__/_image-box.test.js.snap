// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders gallery image Matches the snapshot 1`] = `
<div
  className="makeStyles-box-1"
>
  <div
    className="makeStyles-image-2"
    style={
      Object {
        "backgroundImage": "URL(https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRWXoLhb1QGq1B52D9rNOdWzmIdy0IVkmKk9Q&usqp=CAU)",
      }
    }
  />
  <div
    className="makeStyles-text-3"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      React
    </h6>
  </div>
</div>
`;

exports[`renders gallery image Matches the snapshot admin 1`] = `
<div
  className="makeStyles-box-1"
>
  <div
    className="makeStyles-image-2"
    style={
      Object {
        "backgroundImage": "URL(https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRWXoLhb1QGq1B52D9rNOdWzmIdy0IVkmKk9Q&usqp=CAU)",
      }
    }
  />
  <div
    className="makeStyles-text-3"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      React
    </h6>
  </div>
  <div
    className="makeStyles-buttons-4"
  >
    <div
      className="makeStyles-flexItem-5"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-bttn-9 MuiButton-containedPrimary"
        data-testid="edit-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="Edit"
      >
        <span
          className="MuiButton-label"
        >
          Edit
        </span>
      </button>
    </div>
    <div
      className="makeStyles-flexItem-5"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-flexItem-5 MuiButton-containedSecondary"
        data-testid="delete-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="Delete"
      >
        <span
          className="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</div>
`;
